### PR TITLE
SUS-794 added validation for phalanx filters form

### PR DESCRIPTION
**Pull request by cormorando**
https://wikia-inc.atlassian.net/browse/SUS-794

//cc @rybmat @mixth-sense @macbre

### CODE REVIEW

File: extensions/wikia/PhalanxII/js/SpecialPhalanx.js [hunk 1]
```diff
@@ -30,34 +32,34 @@ require(['jquery', 'mw', 'phalanx', 'BannerNotification'], function($, mw, phala
 
 		// handle "bulk mode" button
 		on('click', '#enterbulk', function(ev) {
```
Comment (WikiaTech) [inline extensions/wikia/PhalanxII/js/SpecialPhalanx.js:34]: MAJOR Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: extensions/wikia/PhalanxII/js/SpecialPhalanx.js [hunk 2]
```diff
@@ -30,34 +32,34 @@ require(['jquery', 'mw', 'phalanx', 'BannerNotification'], function($, mw, phala
...
+			$('#validateMessage').hide();
+			// clear input field when switching modes, for validation purposes
+			filter.val('');
 			// SUS-1191: autofocus newly revealed input field
-			$bulkFilter.focus();
+			bulkFilter.focus();
 		}).
 
 		// handle "single mode" button
 		on('click', '#entersingle', function(ev) {
```
Comment (WikiaTech) [inline extensions/wikia/PhalanxII/js/SpecialPhalanx.js:50]: MAJOR Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: extensions/wikia/PhalanxII/js/SpecialPhalanx.js [hunk 3]
```diff
@@ -76,15 +78,55 @@ require(['jquery', 'mw', 'phalanx', 'BannerNotification'], function($, mw, phala
...
-                selectedOption = $(this).find(':selected');
-
-            if (selectedOption.data('is-custom')) {
-                customExpireField.show().focus();
-            } else {
-                customExpireField.hide();
-            }
-        });
+		// handle custom expire field
+		on('change', '#wpPhalanxExpire', function(ev) {
```
Comment (WikiaTech) [inline extensions/wikia/PhalanxII/js/SpecialPhalanx.js:82]: MAJOR Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: extensions/wikia/PhalanxII/js/SpecialPhalanx.js [hunk 4]
```diff
@@ -30,34 +32,34 @@ require(['jquery', 'mw', 'phalanx', 'BannerNotification'], function($, mw, phala
...
 
+			// clear input field when switching modes, for validation purposes
+			bulkFilter.val('');
 			// SUS-1191: autofocus newly revealed input field
-			$filter.focus();
+			filter.focus();
 		}).
 
 		// handle "validate regex" button
 		on('click', '#validate', function(ev) {
```
Comment (WikiaTech) [inline extensions/wikia/PhalanxII/js/SpecialPhalanx.js:61]: MAJOR Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 12 issues
* MAJOR 6 major
* MINOR 5 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [PhalanxSpecialController.class.php#L295](https://github.com/Wikia/app/blob/bf0bfa79105e0f237a67264ed3da80151e53a7bc/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L295): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [SpecialPhalanx.js#L12](https://github.com/Wikia/app/blob/bf0bfa79105e0f237a67264ed3da80151e53a7bc/extensions/wikia/PhalanxII/js/SpecialPhalanx.js#L12): Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. MINOR [PhalanxSpecialController.class.php#L13](https://github.com/Wikia/app/blob/bf0bfa79105e0f237a67264ed3da80151e53a7bc/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L13): Define a constant instead of duplicating this literal "Phalanx" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PhalanxSpecialController.class.php#L34](https://github.com/Wikia/app/blob/bf0bfa79105e0f237a67264ed3da80151e53a7bc/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L34): Replace this "switch" statement with "if" statements to increase readability. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1301)
1. MINOR [PhalanxSpecialController.class.php#L145](https://github.com/Wikia/app/blob/bf0bfa79105e0f237a67264ed3da80151e53a7bc/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L145): Define a constant instead of duplicating this literal "checkBlocker" 5 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PhalanxSpecialController.class.php#L308](https://github.com/Wikia/app/blob/bf0bfa79105e0f237a67264ed3da80151e53a7bc/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L308): Define a constant instead of duplicating this literal "error" 4 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [PhalanxSpecialController.class.php#L315](https://github.com/Wikia/app/blob/bf0bfa79105e0f237a67264ed3da80151e53a7bc/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L315): Define a constant instead of duplicating this literal "token" 4 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. INFO [PhalanxSpecialController.class.php#L203](https://github.com/Wikia/app/blob/bf0bfa79105e0f237a67264ed3da80151e53a7bc/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L203): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

File: extensions/wikia/PhalanxII/js/SpecialPhalanx.js [hunk 5]
```diff
@@ -76,15 +78,55 @@ require(['jquery', 'mw', 'phalanx', 'BannerNotification'], function($, mw, phala
...
+
+			if (selectedOption.data('is-custom')) {
+				customExpireField.show().focus();
+			} else {
+				customExpireField.hide();
+			}
+		});
+
+	$('#phalanx-block-texts').focusin(function() {
+		$("span[id*='formValidateMessage-filter']").hide();
```
Comment (mszabo-wikia) [inline extensions/wikia/PhalanxII/js/SpecialPhalanx.js:94]: Should we use BannerNotification for validation notices?

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

File: extensions/wikia/PhalanxII/js/SpecialPhalanx.js (hunk 5 above)
Comment (cormorando) [reply extensions/wikia/PhalanxII/js/SpecialPhalanx.js:94]: This is front end validation, regex is done the same way in this form

**Review by mszabo-wikia — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 12 issues
- MAJOR 6 major
- MINOR 5 minor
- INFO 1 info

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [PhalanxSpecialController.class.php#L295](https://github.com/Wikia/app/blob/4c634b6ff2b7dd83516cbe7ded47aea97c5d7811/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L295): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
2. MAJOR [SpecialPhalanx.js#L12](https://github.com/Wikia/app/blob/4c634b6ff2b7dd83516cbe7ded47aea97c5d7811/extensions/wikia/PhalanxII/js/SpecialPhalanx.js#L12): Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
3. MINOR [PhalanxSpecialController.class.php#L13](https://github.com/Wikia/app/blob/4c634b6ff2b7dd83516cbe7ded47aea97c5d7811/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L13): Define a constant instead of duplicating this literal "Phalanx" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
4. MINOR [PhalanxSpecialController.class.php#L34](https://github.com/Wikia/app/blob/4c634b6ff2b7dd83516cbe7ded47aea97c5d7811/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L34): Replace this "switch" statement with "if" statements to increase readability. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1301)
5. MINOR [PhalanxSpecialController.class.php#L145](https://github.com/Wikia/app/blob/4c634b6ff2b7dd83516cbe7ded47aea97c5d7811/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L145): Define a constant instead of duplicating this literal "checkBlocker" 5 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
6. MINOR [PhalanxSpecialController.class.php#L308](https://github.com/Wikia/app/blob/4c634b6ff2b7dd83516cbe7ded47aea97c5d7811/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L308): Define a constant instead of duplicating this literal "error" 4 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
7. MINOR [PhalanxSpecialController.class.php#L315](https://github.com/Wikia/app/blob/4c634b6ff2b7dd83516cbe7ded47aea97c5d7811/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L315): Define a constant instead of duplicating this literal "token" 4 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
8. INFO [PhalanxSpecialController.class.php#L203](https://github.com/Wikia/app/blob/4c634b6ff2b7dd83516cbe7ded47aea97c5d7811/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L203): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

**Pull request merged**